### PR TITLE
[286] - Added CasePreview component

### PR DESCRIPTION
**Pull request by andvirga**
- Added CasePreview component to display Case Information
- DIsplayed Case Info and first connected contact info.
- Added Options Menu
- Added Styling and fixed some unit tests.

### CODE REVIEW

File: plugin-hrm-form/src/components/search/CasePreview/CaseHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+/* eslint-disable react/prop-types */
+import React, { useState } from 'react';
+import { format } from 'date-fns';
+import { Template } from '@twilio/flex-ui';
+import { Button, Menu, MenuItem } from '@material-ui/core';
```
Comment (murilovmachado) [inline plugin-hrm-form/src/components/search/CasePreview/CaseHeader.tsx:5]: Here I suggest using our own **Menu** and **MenuItem**: 
`import { Menu, MenuItem } from '../../menu';`

File: plugin-hrm-form/src/components/search/CasePreview/CaseHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            <Template code="CaseHeader-MergeContactIntoCase" />
+          </BoldText>
+        </MenuItem>
+        <MenuItem onClick={handleClose} style={{ marginTop: '10px' }}>
+          <CopyChildIcon />
+          <BoldText>
+            <Template code="CaseHeader-CopyChildInfo" />
+          </BoldText>
+        </MenuItem>
+      </Menu>
```
Comment (murilovmachado) [inline plugin-hrm-form/src/components/search/CasePreview/CaseHeader.tsx:80]: With using our own **Menu**, we don't need to worry about margins/paddings, it's already handled:
```
import Link from '@material-ui/icons/Link';
import CropFreeIcon from '@material-ui/icons/CropFree';
import CallMergeIcon from '@material-ui/icons/CallMerge';

<Menu anchorEl={anchorEl} open={Boolean(anchorEl)} onClickAway={handleClose}>
    <MenuItem Icon={CropFreeIcon} text={<Template code="CaseHeader-ViewCase" />} onClick={handleClose} />
    <MenuItem
      Icon={CallMergeIcon}
      text={<Template code="CaseHeader-MergeContactIntoCase" />}
      onClick={handleClose}
        />
    <MenuItem Icon={Link} text={<Template code="CaseHeader-CopyChildInfo" />} onClick={handleClose} />
</Menu>
```
(you may notice we're not using the bold text in our Menu, which I think it looks better and stays consistent with what already exists in our app)
Screen Shot 2020-11-16 at 03 11 27

File: plugin-hrm-form/src/components/search/CasePreview/CasePreview.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  const firstContact = connectedContacts && connectedContacts.length > 0 && connectedContacts[0];
+  const { name } = ((firstContact || {}).rawJson || {}).childInformation || {};
+  const { categories, callSummary } = ((firstContact || {}).rawJson || {}).caseInformation || {};
+
+  return (
+    <Flex>
+      <CaseWrapper>
+        <CaseHeader caseId={id} childName={name} createdAt={createdAt} updatedAt={updatedAt} />
+        <CaseSummary summary={callSummary} />
```
Comment (murilovmachado) [inline plugin-hrm-form/src/components/search/CasePreview/CasePreview.tsx:28]: For summary, I'd check if there's a case summary first:
```
const { categories, callSummary } = ((firstContact || {}).rawJson || {}).caseInformation || {};
const summary = currentCase.info?.summary || callSummary;

<CaseSummary summary={summary} />
```

File: plugin-hrm-form/src/utils/categories.ts [hunk 4]
```diff
@@ -57,3 +57,33 @@ export const getContactTags = (contactCategories: ContactCategories) =>
...
+ * @param {{ [category: string]: { [subcategory: string]: boolean } }} categories categories object
+ * @returns {{ [category: string]: string[] }} returns an object containing each truthy subcategory under the category name
+ */
+export const getCaseTags = cats => {
+  if (!cats) return [];
+
+  const mappedCategories = Object.entries(cats).reduce(catsReducer, {});
+
+  return getContactTags(mappedCategories);
+};
```
Comment (murilovmachado) [inline plugin-hrm-form/src/utils/categories.ts:89]: I've realized that we already have this code on **ContactDetailsAdapter.js**. Please take a look at that file, and make it use the code from here or vice versa, it's your call.

File: plugin-hrm-form/src/components/search/CasePreview/CaseTags.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+const renderTag = (tag, color) => (
+  <ContactTag color={color}>
+    <TagMiddleDot color={color} />
+    <TagText color={color}>{tag}</TagText>
+  </ContactTag>
+);
+
+type OwnProps = {
+  categories?: { [category: string]: string[] };
+};
```
Comment (murilovmachado) [inline plugin-hrm-form/src/components/search/CasePreview/CaseTags.tsx:19]: Here, we're getting the categories as they are, so I think the correct type would be:
```
type OwnProps = {
  categories?: {
    [category: string]: {
      [subcategory: string]: boolean;
    };
  };
};
```

**Review by murilovmachado — APPROVED** [review]
Good job! Looks very good. I've added a couple of comments, let me know if you have any questions.

File: plugin-hrm-form/src/components/search/CasePreview/CaseHeader.tsx (hunk 1 above)
Comment (andvirga) [reply plugin-hrm-form/src/components/search/CasePreview/CaseHeader.tsx:5]: Thanks! I didn't notice that this component was used in other pages.

File: plugin-hrm-form/src/components/search/CasePreview/CaseHeader.tsx (hunk 2 above)
Comment (andvirga) [reply plugin-hrm-form/src/components/search/CasePreview/CaseHeader.tsx:80]: Agree!

File: plugin-hrm-form/src/components/search/CasePreview/CasePreview.tsx (hunk 3 above)
Comment (andvirga) [reply plugin-hrm-form/src/components/search/CasePreview/CasePreview.tsx:28]: Good catch!

File: plugin-hrm-form/src/utils/categories.ts (hunk 4 above)
Comment (andvirga) [reply plugin-hrm-form/src/utils/categories.ts:89]: Good catch! I've used that code

**Pull request merged**